### PR TITLE
Run selective fault injection tests in PR gates

### PR DESCRIPTION
**Pull request by abhipsaMisra**

### CODE REVIEW

File: e2e/test/iothub/CombinedClientOperationsPoolAmqpTests.cs [hunk 1]
```diff
@@ -33,7 +33,8 @@ public async Task DeviceSak_DeviceCombinedClientOperations_MultipleConnections_A
                 .ConfigureAwait(false);
         }
 
-        [LoggedTestMethod, Timeout(TestTimeoutMilliseconds)]
+        [LoggedTestMethod, Timeout(LongRunningTestTimeoutMilliseconds)]
+        [TestCategory("LongRunning")]
```
Comment (abhipsaMisra) [inline e2e/test/iothub/CombinedClientOperationsPoolAmqpTests.cs:37]: These tests kept timing out at 3mins intermittently. I increased their timeout to 5mins and marked them "LongRunning".

File: e2e/test/iothub/method/MethodFaultInjectionTests.cs [hunk 2]
```diff
@@ -31,174 +30,172 @@ public class MethodFaultInjectionTests : E2EMsTestBase
...
+                FaultInjection.DefaultFaultDelay).ConfigureAwait(false);
         }
 
+        // Ungraceful disconnection recovery test is marked as a build verification test
+        // to test client reconnection logic in PR runs.
+        [TestCategory("FaultInjectionBVT")]
         [LoggedTestMethod, Timeout(TestTimeoutMilliseconds)]
         public async Task Method_DeviceReceivesMethodAndResponseRecovery_Mqtt()
         {
-            await SendMethodAndRespondRecoveryAsync(new IotHubClientAmqpSettings(),
```
Comment (abhipsaMisra) [inline e2e/test/iothub/method/MethodFaultInjectionTests.cs:55]: this test was incorrectly referencing amqp protocol

File: e2e/test/iothub/method/MethodFaultInjectionTests.cs [hunk 3]
```diff
@@ -31,174 +30,172 @@ public class MethodFaultInjectionTests : E2EMsTestBase
...
+                FaultInjection.FaultType_Tcp,
+                FaultInjection.FaultCloseReason_Boom,
+                FaultInjection.DefaultFaultDelay).ConfigureAwait(false);
         }
 
         [LoggedTestMethod, Timeout(TestTimeoutMilliseconds)]
         public async Task Method_DeviceMethodGracefulShutdownRecovery_MqttWs()
         {
             await SendMethodAndRespondRecoveryAsync(
-                    new IotHubClientAmqpSettings(IotHubClientTransportProtocol.WebSocket),
```
Comment (abhipsaMisra) [inline e2e/test/iothub/method/MethodFaultInjectionTests.cs:66]: this test was incorrectly referencing amqp protocol

File: e2e/test/iothub/method/MethodFaultInjectionTests.cs [hunk 4]
```diff
@@ -31,174 +30,172 @@ public class MethodFaultInjectionTests : E2EMsTestBase
...
         }
 
+        // Ungraceful disconnection recovery test is marked as a build verification test
+        // to test client reconnection logic in PR runs.
+        [TestCategory("FaultInjectionBVT")]
         [LoggedTestMethod, Timeout(TestTimeoutMilliseconds)]
         public async Task Method_DeviceMethodTcpConnRecovery_Amqp()
         {
             await SendMethodAndRespondRecoveryAsync(
-                    new IotHubClientAmqpSettings(),
```
Comment (abhipsaMisra) [inline e2e/test/iothub/method/MethodFaultInjectionTests.cs:77]: tab fix

**Comment by timtay-microsoft** [comment]
Now that they are getting more complex, do you see any value in documenting all these different test categories somewhere? In the ```build.ps1``` script or in a text file in the e2e test folder?

**Review by timtay-microsoft — APPROVED** [review]

**Comment by abhipsaMisra** [comment]
> Now that they are getting more complex, do you see any value in documenting all these different test categories somewhere? In the `build.ps1` script or in a text file in the e2e test folder?

At the risk of making this more complicated than it needs to be, do you see any value in creating an Enum for test categories instead, and referencing that in our tests?

**Comment by timtay-microsoft** [comment]
> > Now that they are getting more complex, do you see any value in documenting all these different test categories somewhere? In the `build.ps1` script or in a text file in the e2e test folder?
> 
> At the risk of making this more complicated than it needs to be, do you see any value in creating an Enum for test categories instead, and referencing that in our tests?

Just for the sake of giving a home to the documentation? I don't see much value in writing an enum for that, no

**Comment by abhipsaMisra** [comment]
> > > Now that they are getting more complex, do you see any value in documenting all these different test categories somewhere? In the `build.ps1` script or in a text file in the e2e test folder?
> > 
> > 
> > At the risk of making this more complicated than it needs to be, do you see any value in creating an Enum for test categories instead, and referencing that in our tests?
> 
> Just for the sake of giving a home to the documentation? I don't see much value in writing an enum for that, no

Not only for the sake of documentation. Instead of having to discover the test category and use the appropriate one for your test, you could select the most appropriate one from this enum. I think it'll make the test categories more discoverable than putting them in a document somewhere.

**Pull request merged**